dragonfly/room2d.py

```diff
@@ -1061,6 +1134,90 @@ def _flip_wall_assigned_objects(original_geo, bcs, win_pars, shd_pars):
         # retrun the flipped lists
         return new_bcs, new_win_pars, new_shd_pars
 
+    def _split_plenum_walls_along_height(self, hb_room, tolerance, floor_height,
+                                         floor_to_ceiling_height):
+        """
+        Args:
+            plenum_hb_room: A plenum represented as a Honeybee Room.
+        """
+        # Set ceiling height
+        ceiling_height = floor_height + floor_to_ceiling_height
+
+        # Iterate through bc of parent room
+        new_faces = [hb_room[0]]
+        for i, bc in enumerate(self._boundary_conditions):
+            face = hb_room[i + 1]
+            if not isinstance(bc, Surface):
+                new_faces.append(face)
+            else:
+                adj_rm = self._parent.room_by_identifier(
+                    bc.boundary_condition_objects[-1])
```

Comment:
I see that you already need for the Room2D to have a parent Story in order for this code to work here. So, in light of this, I would suggest getting rid of the `plenum_ht` input and just raise an exception if `add_plenums` is True and the Room2D has no parent Story.

Reply:
Also, in the case that the Room2D's ceiling height aligns with the top of the story to within the tolerance, no plenum should be generated.